src/Core/Domain/State/QueryResult/EditableState.php

```diff
@@ -56,14 +73,27 @@ class EditableState
 
     /**
      * @param StateId $stateId
+     * @param CountryId $countryId
+     * @param ZoneId $zoneId
      * @param string $name
+     * @param string $isoCode
      * @param bool $enabled
      * @param array $associatedShops
      */
-    public function __construct(StateId $stateId, string $name, bool $enabled, array $associatedShops)
-    {
+    public function __construct(
+        StateId $stateId,
```

Comment:
Usually, the query result objects don't use VOs but only scalar types But you're not the first one who wants to do this I think it's worth deciding this with the team via an ADR